wagtail/admin/tests/test_privacy.py

```diff
@@ -211,6 +211,26 @@ def test_unset_password_restriction(self):
             expected_log_message,
         )
 
+    @override_settings(WAGTAIL_ALLOW_SHARED_PASSWORD_PAGE=False)
+    def test_disable_shared_password_page(self):
+        response = self.client.get(
+            reverse("wagtailadmin_pages:set_privacy", args=(self.public_page.id,)),
+        )
+        self.assertEqual(response.status_code, 200)
+
+        # check that input option for password is not visible
+        self.assertNotIn("password", response.context["form"].fields)
+        self.assertFalse(
+            response.context["form"]
+            .fields["restriction_type"]
+            .valid_value(PageViewRestriction.PASSWORD)
+        )
+
+        # check that the option for password is not visible
+        self.assertNotContains(
+            response, '<div class="w-field__input" data-field-input>'
+        )
```

Comment:
Suggestion: It'd be nice to add this assertion to make sure the input _is_ visible when it's expected. That way we make sure to update this search string should we change it in future.

Reply:
@RealOrangeOne I attempted to test this assertion when the input is visible, but I encountered some difficulties because the response is like the following: <div class=\"w-field__input\" data-field-input>. How can I ensure proper HTML formatting?

Reply:
Either use `html=True` to ignore whitespaces, or use the `get_soup` method from `WagtailTestUtils` to test it with BeautifulSoup, e.g. using `select_one` and a CSS selector

```py
soup = self.get_soup(response.content)
input_el = soup.select_one("[data-field-input]")
self.assertIsNotNone(input_el)
```

(use `assertIsNone` if you need to assert the element isn't there)

Reply:
@laymonage thanks